lib/vsc/utils/rest.py

```diff
@@ -168,9 +169,23 @@ def request(self, method, url, body, headers, content_type=None):
 
         # censor contents of 'Authorization' part of header, to avoid leaking tokens or passwords in logs
         headers_censored = copy.deepcopy(headers)
-        headers_censored['Authorization'] = '<actual authorization header censored>'
+        secret_items = ['Authorization', 'X-Auth-Token']
+        for secret in set(headers_censored).intersection(secret_items):
+            headers_censored[secret] = '<actual authorization header censored>'
+
+        if body and not is_string(body):
+            # censor contents of body to avoid leaking passwords
+            body_censored = copy.deepcopy(body)
+            secret_items = ['password']
+            for secret in set(body_censored).intersection(secret_items):
```

Comment:
this is not safe enough:
```python
>>> set([{'a':'b'}])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'dict'
```

Maybe just wrap it in a 'try' and continue if it fails?

Reply:
fixed in https://github.com/hpcugent/vsc-base/pull/328/commits/b7849d6521a6b89d10c897560d2b37ffab6cb6fc